BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -1458,76 +1417,59 @@ protected override void UpdateImpl()
 
             while (true)
             {
-                RealSense2API.RS2Frame data = RealSense2API.PipelineWaitForFrames(_pipeline, 5000);
-
-                if (!data.IsValid() || data.Handle == IntPtr.Zero)
-                {
-                    RealSense2API.ReleaseFrame(data);
-                    continue;
-                }
+                FrameSet data = _pipeline.WaitForFrames(5000);
 
-                int frameCount = RealSense2API.FrameEmbeddedCount(data);
+                int frameCount = data.Count;
                 log.Debug(string.Format("RealSense2: Got {0} Frames", frameCount));
 
 
                 // extract all frames
-                for (int j = 0; j < frameCount; j++)
+                foreach (VideoFrame vframe in data)
                 {
-                    RealSense2API.RS2Frame frame = RealSense2API.FrameExtract(data, j);
-                    RealSense2API.FrameAddRef(frame);
-
                     // what kind of frame did we get?
-                    RealSense2API.RS2StreamProfile profile = RealSense2API.GetStreamProfile(frame);
-                    RealSense2API.GetStreamProfileData(profile, out RealSense2API.Stream stream, out RealSense2API.Format format, out int index, out int uid, out int framerate);
-
-                    log.Debug(string.Format("RealSense2: Analyzing frame {0}", j + 1));
-                    log.Debug(string.Format("RealSense2: stream {0}", stream.ToString()));
-                    log.Debug(string.Format("RealSense2: format {0}", format.ToString()));
+                    StreamProfile profile = vframe.Profile;
+                    Stream stream = profile.Stream;
+                    Format format = profile.Format;
```

Comment:
Is `format` used? I can't see it in the Web diff view.

Reply:
No, currently it is not. One could use it to check if the format of the frame is actually the specified format of `ActivateChannelImpl`.